erpnext/accounts/doctype/payment_entry/test_payment_entry.py

```diff
@@ -758,6 +758,21 @@ def test_payment_entry_against_onhold_purchase_invoice(self):
 
 		self.assertTrue("is on hold" in str(err.exception).lower())
 
+	def test_payment_entry_over_allocate_amount(self):
+		si = create_sales_invoice()
+
+		pe_draft = get_payment_entry("Sales Invoice", si.name)
+		pe_draft.insert()
+
+		pe = get_payment_entry("Sales Invoice", si.name)
+		pe.submit()
+
+		pe_draft.validate()
```

Comment:
In this case payment entry is getting submitted, since the reference row gets removed and it might be getting allocated as an advance which the users might not want, shouldn't this throw a validation error instead?

Also, you should not call validate method explicitly, it real scenarios the old document would already be validated, it should directly be called from submitting, you should remove the validate call its not needed

Reply:
> In this case payment entry is getting submitted, since the reference row gets removed and it might be getting allocated as an advance which the users might not want, shouldn't this throw a validation error instead?

This is a good point. I will modify.

> Also, you should not call validate method explicitly, it real scenarios the old document would already be validated, it should directly be called from submitting, you should remove the validate call its not needed

Noted, I will  modify.